src/main/java/seedu/address/model/client/ClientName.java

```diff
@@ -0,0 +1,130 @@
+package seedu.address.model.client;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents the name of the client. This class is modelled after the Name class in the Person package of AB3
+ */
+public class ClientName {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Names should not be blank and can contain only contain alphanumeric characters and spaces. A name may "
+                    + "contain at most"
+                    + " "
+                    + "3 words and each word can be at most 10 characters long.";
+
+
+    /**
+     * The name can contain only letters and spaces.
+     */
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+
+    private String fullName;
+
+    /**
+     * Constructs a ClientName.
+     *
+     * @param name A valid name.
+     */
+    public ClientName(String name) {
+        requireNonNull(name);
+        checkArgument(isValidClientName(name), MESSAGE_CONSTRAINTS);
+        fullName = name;
+    }
+
+    /**
+     * Represents an Empty ClientName.
+     */
+    private static class EmptyClientName extends ClientName {
+        private static final ClientName EMPTY_NAME = new EmptyClientName();
+        private EmptyClientName() {
+            super("");
+        }
+
+        /**
+         * Checks if this Client Name is empty.
+         * @return true if the Client Name is empty.
+         */
+        @Override
+        public boolean isEmpty() {
+            return true;
+        }
+    }
+
+
+    /**
+     * Returns true if a given string is a valid name. A name is valid only if it contains oly letters and spaces and
+     * had a maximum of three words, each of length less than 10 characters.
+     * @param test String representing name to be tested
+     * @return boolean true if a given string is a valid name
+     */
+    public static boolean isValidClientName(String test) {
+
+        String[] words = test.split(" ");
```

Comment:
Regex would be a cleaner way to check the validity of the name; the default checking for `Name` would suffice however

Reply:
Have an extra check for `ClientName` with respect to the number of words and so couldn't use `name`/inherit from it without violating LSP. However, we can abstract out all projectName, clientName, and other repetitive files as part of the next iteration